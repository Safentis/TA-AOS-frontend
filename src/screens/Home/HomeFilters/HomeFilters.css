.filters__label {
    margin-right: 1.5rem;
}

.filters__select {
    width: 10rem;
}