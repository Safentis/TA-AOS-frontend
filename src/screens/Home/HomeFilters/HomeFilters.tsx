import React, { FC } from 'react';
import { Select, Input } from 'antd';

import Label from '../../../components/Label/Label';

import './HomeFilters.css';
import { Props } from './HomeFilters.interface';
import { Filter } from '../../../App/App.interface';
import { useGlobalContext } from '../../../App/App.context';

const HomeFilters: FC<Props> = ({
  className = '',
}): React.ReactElement => {
  const { Option } = Select;
  const { filter, setFilter, handlePagination } = useGlobalContext();

  //* ------------------------------------------------------
  //* Hendlers of state
  const handleSearch = (event: React.ChangeEvent): void => {
    const newState: Filter = { ...filter };

    //* In this case, we are getting value of input
    const target = event.target as HTMLInputElement;
    const value: string = target.value;

    //* Set new value to state
    newState.search = value;

    //* After we set new state
    setFilter(newState);
    handlePagination(1);
  };

  const handleSelect = (value: string, options: any): void => {
    const newState: Filter = { ...filter };
    const selectCategory: string = options.title;
    const valueOfSelect: string = value !== 'none' ? value : '';
    //* In this case, we are setting value of select
    //* in newState object
    if (selectCategory === 'status') {
      newState.status = valueOfSelect;
    } else {
      newState.gender = valueOfSelect;
    }

    setFilter(newState);
    handlePagination(1);
  };

  //* ------------------------------------------------------
  //* Content
  const status: string[] = ['alive', 'dead', 'unknown', 'none'];
  const gender: string[] = [
    'male',
    'female',
    'genderless',
    'unknown',
    'none',
  ];

  const STATUS_OPTIONS = status.map((value: string) => (
    <Option key={value} value={value} title="status">
      {value}
    </Option>
  ));

  const GENDER_OPTIONS = gender.map((value: string) => (
    <Option key={value} value={value} title="gender">
      {value}
    </Option>
  ));

  return (
    <div className={`filters ${className}`}>
      <Label className="filters__label" content="search">
        <Input
          className="search filters__search"
          placeholder="Type here..."
          onChange={handleSearch}
          id="search"
          value={filter.search}
        />
      </Label>

      <Label className="filters__label" content="status">
        <Select
          className="select filters__select"
          onChange={handleSelect}
          placeholder="Status"
          defaultValue="none"
        >
          {STATUS_OPTIONS}
        </Select>
      </Label>

      <Label className="filters__label" content="gender">
        <Select
          className="select filters__select"
          onChange={handleSelect}
          placeholder="Gender"
          defaultValue="none"
        >
          {GENDER_OPTIONS}
        </Select>
      </Label>
    </div>
  );
};

export default HomeFilters;
