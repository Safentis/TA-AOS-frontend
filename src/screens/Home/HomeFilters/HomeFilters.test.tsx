import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import HomeFilters from './HomeFilters';

import { Props } from './HomeFilters.interface';

const setUp = (props: Props) => {
  return shallow(<HomeFilters {...props} />);
};

describe('<HomeFilters />', () => {
  let component: ShallowWrapper<
    any,
    Readonly<{}>,
    React.Component<{}, {}, any>
  >;
  let props: Props = {
    className: 'custom-style',
  };

  beforeEach(() => {
    component = setUp(props);
  });

  it('component does match to the snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  describe('Markup', () => {
    it('has 1 Input element', () => {
      expect(component.find('Input')).toBeTruthy();
    });
  });
});
