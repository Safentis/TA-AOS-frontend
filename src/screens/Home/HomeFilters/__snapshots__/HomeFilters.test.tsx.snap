// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeFilters /> component does match to the snapshot 1`] = `
<div
  className="filters custom-style"
>
  <Label
    className="filters__label"
    content="search"
  >
    <Input
      className="search filters__search"
      id="search"
      onChange={[Function]}
      placeholder="Type here..."
      type="text"
      value=""
    />
  </Label>
  <Label
    className="filters__label"
    content="status"
  >
    <ForwardRef(InternalSelect)
      className="select filters__select"
      defaultValue="none"
      onChange={[Function]}
      placeholder="Status"
    >
      <Option
        key="alive"
        title="status"
        value="alive"
      >
        alive
      </Option>
      <Option
        key="dead"
        title="status"
        value="dead"
      >
        dead
      </Option>
      <Option
        key="unknown"
        title="status"
        value="unknown"
      >
        unknown
      </Option>
      <Option
        key="none"
        title="status"
        value="none"
      >
        none
      </Option>
    </ForwardRef(InternalSelect)>
  </Label>
  <Label
    className="filters__label"
    content="gender"
  >
    <ForwardRef(InternalSelect)
      className="select filters__select"
      defaultValue="none"
      onChange={[Function]}
      placeholder="Gender"
    >
      <Option
        key="male"
        title="gender"
        value="male"
      >
        male
      </Option>
      <Option
        key="female"
        title="gender"
        value="female"
      >
        female
      </Option>
      <Option
        key="genderless"
        title="gender"
        value="genderless"
      >
        genderless
      </Option>
      <Option
        key="unknown"
        title="gender"
        value="unknown"
      >
        unknown
      </Option>
      <Option
        key="none"
        title="gender"
        value="none"
      >
        none
      </Option>
    </ForwardRef(InternalSelect)>
  </Label>
</div>
`;
